alternator/stats.cc

```diff
@@ -89,47 +88,144 @@ stats::stats() : api_operations{} {
             seastar::metrics::make_counter("batch_item_count", seastar::metrics::description("The total number of items processed across all batches"),{op("BatchWriteItem")},
-                    api_operations.batch_write_item_batch_total)(alternator_label).set_skip_when_empty(),
+                    stats.api_operations.batch_write_item_batch_total)(alternator_label).set_skip_when_empty(),
             seastar::metrics::make_counter("batch_item_count", seastar::metrics::description("The total number of items processed across all batches"),{op("BatchGetItem")},
-                    api_operations.batch_get_item_batch_total)(alternator_label).set_skip_when_empty(),
+                    stats.api_operations.batch_get_item_batch_total)(alternator_label).set_skip_when_empty(),
             seastar::metrics::make_histogram("batch_item_count_histogram", seastar::metrics::description("Histogram of the number of items in a batch request"),{op("BatchGetItem")},
-                    [this]{ return estimated_histogram_to_metrics(api_operations.batch_get_item_histogram);})(alternator_label).aggregate({seastar::metrics::shard_label}).set_skip_when_empty(),
+                    [&stats]{ return estimated_histogram_to_metrics(stats.api_operations.batch_get_item_histogram);})(alternator_label).aggregate({seastar::metrics::shard_label}).set_skip_when_empty(),
             seastar::metrics::make_histogram("batch_item_count_histogram", seastar::metrics::description("Histogram of the number of items in a batch request"),{op("BatchWriteItem")},
-                    [this]{ return estimated_histogram_to_metrics(api_operations.batch_write_item_histogram);})(alternator_label).aggregate({seastar::metrics::shard_label}).set_skip_when_empty(),
+                    [&stats]{ return estimated_histogram_to_metrics(stats.api_operations.batch_write_item_histogram);})(alternator_label).aggregate({seastar::metrics::shard_label}).set_skip_when_empty(),
     });
 }
+static seastar::metrics::label column_family_label("cf");
+static seastar::metrics::label keyspace_label("ks");
 
+void register_metrics(seastar::metrics::metric_groups& metrics, stats& stats, const sstring& ks, const sstring& table) {
+    // Register the
+    seastar::metrics::label op("op");
+    auto cf_lable = column_family_label(table);
+    auto ks_lable = keyspace_label(ks);
+    metrics.add_group("alternator", {
+#undef OPERATION
+#define OPERATION(name, CamelCaseName) \
+                seastar::metrics::make_total_operations("table_operation", stats.api_operations.name, \
+                        seastar::metrics::description("number of operations via Alternator API"), {op(CamelCaseName), alternator_label, basic_level, cf_lable, ks_lable}).aggregate({seastar::metrics::shard_label}).set_skip_when_empty(),
+#undef OPERATION_LATENCY
+#define OPERATION_LATENCY(name, CamelCaseName) \
```

Comment:
I don't understand what I'm seeing here. Why do you need another duplicate of this seemingly identical registration?
Or maybe there are small differences?

I'm worried that in the future someone will add a metric to one list and forget to add it to the other list. Could we make this somehow a single function perhaps with a boolean parameter "per_table" and use that parameter where there is a difference between the code for per-table and not per-table?

Reply:
I will